recipes/configs/dev/qwen3B_async_grpo.yaml

```diff
@@ -10,144 +10,103 @@
-  ]
-  recipe_checkpoint: null
-  output_dir: ${output_dir}
-  model_type: QWEN2
-
-
-ref_checkpointer:
-  _component_: torchtune.training.FullModelHFCheckpointer
-  checkpoint_dir: ${base_model_path}
-  checkpoint_files: [
-    model-00001-of-00002.safetensors,
-    model-00002-of-00002.safetensors,
-  ]
-  recipe_checkpoint: null
-  output_dir: ${output_dir}/ref  # shouldn't be used?
-  model_type: QWEN2
-
-
-inference:
-  engine: vllm
-  top_k: null
-  temperature: 1.0
-  tp_size: 1
-  max_generated_tokens: 512
-  batch_size: 1
-  group_size: 16
-  total_batch_size: ${eval:'${inference.batch_size} * ${inference.group_size}'}
-  steps_before_sync: 1
-  queue_maxsize: ${eval:'${orchestration.num_inference_workers} * ${training.steps_before_sync}'}
-
-postprocessing:
-  device_type: cuda
-  dtype: bf16
+# All orchestration args
+orchestration:
+  num_inference_workers: 4
+  num_postprocessing_workers: 1
+  num_training_workers: 2
+  replay_buffer_size: ${inference.batch_size}  # TODO: Right now this can't be bigger, or else we'll get padding issues
+  num_steps: 250
 
+# All training args
 training:
+  checkpointer:
+    _component_: torchtune.training.FullModelHFCheckpointer
+    checkpoint_dir: ${base_model_path}
+    checkpoint_files: [
+      model-00001-of-00002.safetensors,
+      model-00002-of-00002.safetensors,
+    ]
+    recipe_checkpoint: null
+    output_dir: ${output_dir}
+    model_type: QWEN2
   batch_size: 16
-  epochs: 10
   ppo_epochs: 1
   clip_grad_norm: 1.0
-  device_type: cuda
-  resume_from_checkpoint: False
-  save_every_n_epochs: 1
-
-  # Memory management
+  save_every_n_steps: 250
   enable_activation_checkpointing: True  # True reduces memory
   enable_activation_offloading: True  # True reduces memory.
-  compile: False  # pytorch compile, set to true for better perf/memory
-
-  dtype: bf16
-
-  # number of train_steps after which to do weight sync
-  steps_before_sync: 2
-
+  compile: False  # torch.compile, set to true for better perf/memory
+  steps_before_weight_sync: 2
   optimizer:
     _component_: torch.optim.AdamW
     lr: 1e-5
     fused: True
-  lr_scheduler:
-    _component_: torchtune.training.lr_schedulers.get_cosine_schedule_with_warmup
-    num_warmup_steps: 50
   loss:
     _component_: torchtune.dev.grpo.loss.GRPOWithChunkedOutputLoss
     kl_coeff: 0.01
     epsilon: 0.2
+  seed: null
 
-orchestration:
-  num_inference_workers: 4
-  num_postprocessing_workers: 1
-  num_training_workers: 2
-  replay_buffer_size: ${inference.batch_size}  # TODO this needs to be fixed. Right now this can't be bigger, or else we'll get padding issues
-  num_steps: 250
+# All post-processing args
+postprocessing:
+  ref_checkpointer:
+    _component_: torchtune.training.FullModelHFCheckpointer
+    checkpoint_dir: ${base_model_path}
+    checkpoint_files: [
+      model-00001-of-00002.safetensors,
+      model-00002-of-00002.safetensors,
+    ]
+    model_type: QWEN2
+
+# All inference args
+inference:
```

Comment:
Nit: maybe we should put inference at the top? I think conceptually that's how the recipe flows: generation, postprocessing, training, weight update